### ISSUE TITLE
patch for compare.py - exit code

### ISSUE THREAD

**Issue by dominicraf** [rdiff-backup/rdiff-backup#1]
I have a patch (from compare.py in rdiff-backup 1.2.8) below which I use (and need) so that if a verification run fails with one or more verification hash failures (as opposed to some other sort of glitch/crash) we get a specific exit number (2), and a specific message. The original code gives an exit number equal to the total of verification failures and no specific text - this makes it impossible to be sure of the cause of failure - not least because if the run is aborted (killed) the returned code is 1. Please can this patch be incorporated in future releases:

```
--- compare.py  2015-11-30 17:12:56.046174495 +0000
+++ compare.py  2015-11-30 17:16:23.978174495 +0000
@@ -101,2 +101,5 @@
-       if not bad_files: log.Log("Every file verified successfully.", 3)
-       return bad_files
+       if bad_files:
+               log.Log("Not all files could be verified.", 3)
+               return 2
+       log.Log("Every file verified successfully.", 3)
+       return 0
```

**Comment by davekempe** [comment]
congrats on issue #1. Wes will review it and likely apply it once we have the tree worked out. It could be time for a bit of a re-org first.

**Comment by anomaly256** [comment]
I agree that exit code values should have specific error condition meaning.  This is convention almost everywhere else.  Obviously some consumers of rdiff-backup will need to be updated to account for such a functionality change though.  Unless someone can give a compelling argument against this I think we will adopt it.  If it proves too polarizing we can consider an 'enable old behavior' command line argument.

**Comment by szpak** [comment]
As the code is now hosted on GitHub it may be probably a good idea to switch to [pull request](https://help.github.com/articles/using-pull-requests/) based collaboration model (instead of patches).

**Comment by davekempe** [comment]
Agreed szpak, but we will take either for now. There are a bunch of patches people have come up with  that need merging anyway.

**Comment by ottok** [comment]
Hello @dominicraf !

The code base has been rewritten for Python 3 and there will be no further releases for the 1.x series in Python 2 anymore. Could you please help out and test again with the latest development version (https://github.com/rdiff-backup/rdiff-backup#installing-the-latest-development-version) if you can still reproduce this?

With your help, the upcoming 2.0.0 release could be improved and this reported issue eventually solved in future 2.x releases.

**Comment by ericzolf** [comment]
This is a bigger change which is necessary: not only Verify does return the number of failures as return code, also all the Compare functions do this. This said, I fully agree that it isn't best practice and requires a fix. It'll be done as part of milestone 2.2.0.

Some details of what needs to be done:

- modify all `def Compare*` functions and the `def Verify` function in `compare.py`
- make sure they return a fixed value (2 is fine with me) instead of the number of files
- make sure they output in failure case a message of level 1 stating how many files didn't pass the comparison/verification (if rdiff-backup returns an error code, there must be an error message).

I mark the bug as beginner as someone new can do it.

**Comment by ericzolf** [comment]
Closing as this has been addressed by #343 after more than 4 years!
@dominicraf thanks for your patience. And congratulation on closing yourself with your PR your own issue #1
